game-core/src/main/java/games/strategy/triplea/delegate/battle/casualty/AaCasualtySelector.java

```diff
@@ -37,6 +37,7 @@ public static CasualtyDetails getAaCasualties(
       final Collection<Unit> allFriendlyUnits,
       final Collection<Unit> defendingAa,
       final Collection<Unit> allEnemyUnits,
+      final String text,
```

Comment:
I don't know if pushing this complexity to the caller is a good move. The caller will have to know the full implementation details of this method to know how this text is being used.

I think for example one smell is the value of parameter is a composition of other parameters:
```
            "Select "
                + dice.getHits()
                + " casualties from "
                + currentTypeAa
                + " fire in "
                + territory.getName(),
```

- Perhaps we can fix this cleanly with a configuration flag? 

To another extent, the root cause seems to be the overall structure here and arguably a lack of abstractions altogether. I think this is a good example of a "routine", in other words a procedural method. The deeper fix here is likely to be quite the rabbit hole.

Reply:
As I mentioned in the description, the "battleid" is basically used as a configuration flag.  It controls whether the entire text is used as is or if it will append ", select X casualties".

We are limited on what changes we can make because this goes through a RemoteAction.  I looked at a possibility of change and one idea is to only pass the `currentTypeAa` (or similar for non aa) through the text parameter.  In the final callees (`BattleDisplay#getCasualties` and `BattlePanel#getCasualtiesAa`), they can get access to the territory and the dice.  So at that point, they can generate the final string.

One potential issue is that `AirBattle.AttackersFire#execute` and `AirBattle.AttackersFire#execute` use a different string format: "Defenders Fire" and "Attackers Fire".  By making this change, they would now say something like "Hits from attackers" and "Hits from defenders".

Reply:
TBH I really cringe by passing a UI specific text to this API, but I recognize the code we're working with is tangled into knots. Are there any other reasonable options, or is this really the least worst?

Reply:
Considering we can't add additional parameters or change the types of the parameters, I think either leaving it as is or making the change to only passing the `currentTypeAa` and similar is what we have available.

Reply:
:+1: